astropy/time/formats.py

```diff
@@ -691,6 +691,156 @@ def to_value(self, timezone=None, parent=None):
     value = property(to_value)
 
 
+class TimeYMDHMS(TimeUnique):
+    """
+    ymdhms: A Time format to represent Time as year, month, day, hour,
+    minute, second (thus the name ymdhms)
+
+    Acceptable inputs must have keys or column names in the "YMDHMS" set of
+    ``year``, ``month``, ``day`` ``hour``, ``minute``, ``second``.
+
+    - Dict with keys in the YMDHMS set
+    - List of dict with keys in the YMDHMS set
+    - NumPy structured array, record array or astropy Table, or single row
+      of those types, with column names in the YMDHMS set
+
+    Example::
+
+      >>> from astropy.time import Time
+      >>> t = Time({'year': [2015, 2016], 'minute': [1, 2]}, scale='utc')
+      >>> t.iso
+      array(['2015-01-01 00:01:00.000', '2016-01-01 00:02:00.000'], dtype='<U23')
+      >>> t.ymdhms.year
+      array([2015, 2016], dtype=int32)
+    """
+    name = 'ymdhms'
+
+    def _check_val_type(self, val1, val2):
+        """
+        This checks inputs for the YMDHMS format.
+
+        It is bit more complex than most format checkers because of the flexible
+        input that is allowed.  Also, it actually coerces ``val1`` into an appropriate
+        dict of ndarrays that can be used easily by ``set_jds()``.  This is useful
+        because it makes it easy to get default values in that routine.
+
+        Parameters
+        ----------
+        val1 : ndarray or None
+        val2 : ndarray or None
+
+        Returns
+        -------
+        val1_as_dict, val2 : val1 as dict or None, val2 is always None
+
+        """
+        if val2 is not None:
+            raise ValueError('val2 must be None for ymdhms format')
+
+        ymdhms = ['year', 'month', 'day', 'hour', 'minute', 'second']
+
+        if val1.dtype.names:
+            # NumPy structured array or row thereof
+            if not set(val1.dtype.names).issubset(set(ymdhms)):
```

Comment:
Is this strict enough? I'd think that if `day` is present, `month` should be present as well, i.e., no holes.